backend/BudgetMate/data_ingestion/models.py

```diff
@@ -61,7 +71,7 @@ class StatementLine(models.Model):
         on_delete=models.CASCADE,
     )
     operation_type = models.CharField(max_length=255, choices=OperationType.choices)
-    amount = models.IntegerField()
+    amount = models.FloatField()
```

Comment:
I suggest we use [DecimalField](https://docs.djangoproject.com/en/4.2/ref/models/fields/#decimalfield) instead.

floats can have funny behaviors : 
```python
Python 3.10.9 (main, Feb 20 2023, 15:58:28) [GCC 11.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: 0.1 + 0.2
Out[1]: 0.30000000000000004
```
This is because floats use binary representation that make storing some numbers a bit tricky

Decimal on the other hand has a sense of significant numbers : 

```py
Python 3.10.9 (main, Feb 20 2023, 15:58:28) [GCC 11.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from decimal import Decimal

In [2]: from decimal import getcontext

In [3]: getcontext().prec = 2

In [4]: Decimal(0.1) + Decimal(0.2)
Out[4]: Decimal('0.30')
```

In this app we're going to deal with amounts that are limites to a 2 digits precision (centimes), we wouldn't want the total of spendings to be looking like 1203.30000000000000004 right ?

Reply:
Done !